### PR TITLE
Create class for person, students and teachers

### PR DESCRIPTION
**Pull request by JoyTaribagshaw**
In this branch, I implemented the following;
>

- [ ] Create class Person with the following:

- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- [ ] Create class Student with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".
- Create class Teacher with the following:
- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.

- [ ] Each class is saved in a separate file.

### CODE REVIEW

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @JoyTaribagshaw, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a readme file with a Live Demo section that has no Live link. If you have no live link to share kindly remove the section.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1 +1,133 @@
...
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Setup](#setup)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
```
Comment (Munsa1) [inline README.md]: - [ ] Kindly remove these broken links as they point to parts of the readme file that have been removed. Please remove the `Install` link as well

File: README.md [hunk 2]
```diff
@@ -1 +1,133 @@
...
+<!-- AUTHORS -->
+
+## 👥 Authors <a name="authors"></a>
+
+👤 *Joy Tari*
+
+- GitHub: [Joy](https://github.com/JoyTaribagshaw)
+- Twitter: [Joy](https://twitter.com/JoyTariBagshaw)
+- LinkedIn: [Joy](www.linkedin.com/in/joytaribagshaw)
+
```
Comment (Munsa1) [inline README.md]: - [ ] Kindly fix the broken link to the LinkedIn profile as it is showing a GitHub 404 page.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
```
Comment (Munsa1) [inline person.rb:19]: - [ ] Kindly make the method `of_age?` private as per requirement.

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require_relative 'person'
+
+# class for student
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(age, classroom, name = 'unknown')
+    super(age, name)
+    @classroom = classroom
+  end
```
Comment (Munsa1) [inline student.rb]: - [ ] Kindly add a parent permission parameter to the initialize method as per requirement.

Code example
```ruby
def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+require_relative 'person'
+
+# class for teacher
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name = 'unknown')
+    super(age, name)
+    @specialization = specialization
+  end
```
Comment (Munsa1) [inline teacher.rb]: - [ ] Again here kindly add the parent permission parameter to your initialize method as per requirement.

Code example
```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @JoyTaribagshaw, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Created Classes In Separate Files :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive Pr :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (JoyTaribagshaw) [reply person.rb:19]: Thank you for your review, I have made the changes required.

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @JoyTaribagshaw 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Implement classes and objects in Ruby.
- ✔️ Implement encapsulation and inheritance with Ruby.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**